### PR TITLE
implement isLocal for shared storage

### LINKED ISSUE #20484: Orientation for shared images broken
After upgrading ownCloud to version 8.1.4, shared images (including thumbnails) that require rotation are no longer being rotated in the UI. This is only happening when an account other than the owner of the image is accessing it. The owner can still see all the thumbnails and images with a correct orientation based on EXIF data.

Example of debug entries in the log when one of these shared images is accessed:

    {"reqId":"8+4ddYHXwnHf3I\/TBn+e","remoteAddr":"192.168.1.2","app":"core","message":"Generating preview for \"\/shared folders\/photos\/Camera New (test)\/20151108_115921.jpg\" with \"OC\\Preview\\JPEG\"","level":0,"time":"November 12, 2015 07:06:22","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fshared+folders%2Fphotos%2FCamera+New+%28test%29%2F20151108_115921.jpg&c=3d7aac3fcadb160785927bd393424122&x=36&y=36&forceIcon=0"}    
    {"reqId":"x5k0x0ZoC2s7vld9Qqa1","remoteAddr":"192.168.1.2","app":"core","message":"OC_Image->fixOrientation() No readable file path set.","level":0,"time":"November 12, 2015 07:06:22","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fshared+folders%2Fphotos%2FCamera+New+%28test%29%2F20151111_110734.jpg&c=0337a9dedd0e4906f82906be8fd4bf45&x=36&y=36&forceIcon=0"}
    {"reqId":"8+4ddYHXwnHf3I\/TBn+e","remoteAddr":"192.168.1.2","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"November 12, 2015 07:06:23","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fshared+folders%2Fphotos%2FCamera+New+%28test%29%2F20151108_115921.jpg&c=3d7aac3fcadb160785927bd393424122&x=36&y=36&forceIcon=0"}

The problem seems to be caused by a change in 'lib\private\preview\image.php'. Here's the diff between the code in 8.1.3 and 8.1.4:

    49c49,50
    < 		if ($fileInfo['encrypted'] === true) {
    ---
    > 		$useTempFile = $fileInfo->isEncrypted() || !$fileInfo->getStorage()->isLocal();
    > 		if ($useTempFile) {
    54a56,58
    > 		if ($useTempFile) {
    > 			unlink($fileName);
    > 		}

I have no external storage and no encryption configured on my server, but for some reason, the shared file path becomes something like '/tmp/oc_tmp_1SPFjp-.JPG', which causes the EXIF info lookup to fail. I didn't know what the right fix would be here since I'm not familiar with the "larger picture" but reverting to the code snippet above fixed my problem. Just raising this in case anyone else notices this issue. All this was happening on an otherwise unchanged CentOS 6 / Apache 2.2 / PHP 5.6 server with ownCloud running on it for over a year and a half without issues.

### PR DESCRIPTION
**Pull request by icewind1991**
Fixes https://github.com/owncloud/core/issues/20484

cc @PVince81 @rullzer

### CODE REVIEW

**Comment by PVince81** [comment]
Code makes sense :+1:

File: apps/files_sharing/lib/sharedstorage.php [hunk 1]
```diff
@@ -699,4 +699,11 @@ public function getAvailability() {
 	public function setAvailability($available) {
 		// shares do not participate in availability logic
 	}
+
+	public function isLocal() {
```
Comment (rullzer) [inline apps/files_sharing/lib/sharedstorage.php]: missing phpdoc

**Comment by rullzer** [comment]
Beisde the missing phpdoc it makes sense :+1:

File: apps/files_sharing/lib/sharedstorage.php (hunk 1 above)
Comment (icewind1991) [inline apps/files_sharing/lib/sharedstorage.php]: phpdoc is in the interface this implements

Comment (rullzer) [inline apps/files_sharing/lib/sharedstorage.php]: inherit doc then?

But ok ;)

**Comment by MorrisJobke** [comment]
@karlitschek Backport of this to stable8.2 was requested in https://github.com/owncloud/core/issues/20484#issuecomment-206473121

**Comment by karlitschek** [comment]
makes sense. please backport

**Comment by MorrisJobke** [comment]
> @karlitschek Backport of this to stable8.2 was requested in #20484 (comment)

@icewind1991 Could I ask you to do the backport PR? Thanks

**Comment by icewind1991** [comment]
8.2: https://github.com/owncloud/core/pull/24000

**Pull request merged**